File: domain/src/main/java/com/iamkurtgoz/domain/model/request/RegisterRequest.kt
/*
 * Copyright 2024 Sporthor Android
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.iamkurtgoz.domain.model.request

import android.os.Parcelable
import androidx.annotation.Keep
import kotlinx.parcelize.Parcelize
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Keep
@Serializable
data class RegisterRequest(
    @SerialName("email") val email: String?,
    @SerialName("firebaseId") val firebaseId: String?,
    @SerialName("firebaseToken") val firebaseToken: String?,
    @SerialName("mobilePhone") val mobilePhone: String?,
    @SerialName("name") val name: String?,
    @SerialName("password") val password: String?,
    @SerialName("socialInfo") val socialInfo: RegisterSocialInfoRequest?,
    @SerialName("surname") val surname: String?,
    @SerialName("username") val username: String?,
)

@Keep
@Serializable
@Parcelize
data class RegisterSocialInfoRequest(
    @SerialName("accessToken") val accessToken: String?,
    @SerialName("accountId") val accountId: String?,
    @SerialName("email") val email: String?,
    @SerialName("platform") val platform: Int?,
) : Parcelable

File: Sporthor/Sporthor/ViewControllers/Authentication/Models/RegisterRequest.swift
//
//  RegisterRequest.swift
//  Sporthor
//
//  Created by derTurke on 1.03.2025.
//

import Foundation

struct RegisterRequest: Encodable {
    var name: String
    var surname: String
    var email: String
    var username: String
    var mobilePhone: String
    var password: String
    var firebaseToken: String
    var firebaseId: String
    var socialInfo: SocialInfo?
    
    init(name: String = "",
         surname: String = "",
         email: String = "",
         username: String = "",
         mobilePhone: String = "",
         password: String = "",
         firebaseToken: String = "",
         firebaseId: String = "",
         socialInfo: SocialInfo? = nil) {
        self.name = name
        self.surname = surname
        self.email = email
        self.username = username
        self.mobilePhone = mobilePhone
        self.password = password
        self.firebaseToken = firebaseToken
        self.firebaseId = firebaseId
        self.socialInfo = socialInfo
    }
    
    enum CodingKeys: CodingKey {
        case name
        case surname
        case email
        case username
        case mobilePhone
        case password
        case firebaseToken
        case firebaseId
        case socialInfo
    }
    
    func encode(to encoder: any Encoder) throws {
        var container = encoder.container(keyedBy: CodingKeys.self)
        try container.encode(self.name, forKey: .name)
        try container.encode(self.surname, forKey: .surname)
        try container.encode(self.email, forKey: .email)
        try container.encode(self.username, forKey: .username)
        try container.encode(self.mobilePhone, forKey: .mobilePhone)
        try container.encode(self.password, forKey: .password)
        try container.encode(self.firebaseToken, forKey: .firebaseToken)
        try container.encode(self.firebaseId, forKey: .firebaseId)
        try container.encode(self.socialInfo, forKey: .socialInfo)
    }
}
